src/services/codefixes/addMissingAwait.ts

```diff
@@ -0,0 +1,175 @@
+/* @internal */
+namespace ts.codefix {
+    type ContextualTrackChangesFunction = (cb: (changeTracker: textChanges.ChangeTracker) => void) => FileTextChanges[];
+    const fixId = "addMissingAwait";
+    const propertyAccessCode = Diagnostics.Property_0_does_not_exist_on_type_1.code;
+    const callableConstructableErrorCodes = [
+        Diagnostics.This_expression_is_not_callable.code,
+        Diagnostics.This_expression_is_not_constructable.code,
+    ];
+    const errorCodes = [
+        Diagnostics.An_arithmetic_operand_must_be_of_type_any_number_bigint_or_an_enum_type.code,
+        Diagnostics.The_left_hand_side_of_an_arithmetic_operation_must_be_of_type_any_number_bigint_or_an_enum_type.code,
+        Diagnostics.The_right_hand_side_of_an_arithmetic_operation_must_be_of_type_any_number_bigint_or_an_enum_type.code,
+        Diagnostics.Operator_0_cannot_be_applied_to_type_1.code,
+        Diagnostics.Operator_0_cannot_be_applied_to_types_1_and_2.code,
+        Diagnostics.This_condition_will_always_return_0_since_the_types_1_and_2_have_no_overlap.code,
+        Diagnostics.Type_0_is_not_an_array_type.code,
+        Diagnostics.Type_0_is_not_an_array_type_or_a_string_type.code,
+        Diagnostics.Type_0_is_not_an_array_type_or_a_string_type_Use_compiler_option_downlevelIteration_to_allow_iterating_of_iterators.code,
+        Diagnostics.Type_0_is_not_an_array_type_or_a_string_type_or_does_not_have_a_Symbol_iterator_method_that_returns_an_iterator.code,
+        Diagnostics.Type_0_is_not_an_array_type_or_does_not_have_a_Symbol_iterator_method_that_returns_an_iterator.code,
+        Diagnostics.Type_0_must_have_a_Symbol_iterator_method_that_returns_an_iterator.code,
+        Diagnostics.Type_0_must_have_a_Symbol_asyncIterator_method_that_returns_an_async_iterator.code,
+        Diagnostics.Argument_of_type_0_is_not_assignable_to_parameter_of_type_1.code,
+        propertyAccessCode,
+        ...callableConstructableErrorCodes,
+    ];
+
+    registerCodeFix({
+        fixIds: [fixId],
+        errorCodes,
+        getCodeActions: getAllPossibleFixesForError,
+        getAllCodeActions: context => {

```

Comment:
why inline getAllCodeActions but not getCodeActions? They should probably be the same.

(I'd vote for both inline, but don't care too much.)

Reply:
I think there was a transient reason which later got refactored away. Good call.